### ISSUE TITLE
Enhance documentation/commenting of refine methods

### ISSUE THREAD

**Issue by tc85324** [gchq/coreax#585]
### What's the new feature?

Provided clearer documentation of what exactly the refine methods are doing, and how their "comparison` computations work. For example, there is no documentation/comment explaining where this calculation has come from, what precisely it is computing, and/or if it needs to be normalized (as it is being used to find an argmax).
```python
return (
            kernel.compute(x[coreset_indices], x[i]).sum()
            - kernel.compute(x[candidate_indices, :], x[coreset_indices]).sum(axis=1)
            + kernel.compute(x[candidate_indices, :], x[i])[:, 0]
            - kernel_gram_matrix_diagonal[candidate_indices]
        ) / (num_points_in_coreset**2) - (
            kernel_matrix_row_sum_mean[i]
            - kernel_matrix_row_sum_mean[candidate_indices]
        ) / num_points_in_coreset
```

### What value does this add?

Enhances code clarity and enables future developers to suggest alternative (potentially more efficient) computations.

**Comment by tp832944** [comment]
@tc85324 what is blocking this? Please tag the relevant tickets in the description or as a comment.

**Comment by tp832944** [comment]
Probably addressed by reduction refactor #581.

**Comment by pc532627** [comment]
Closed by #581
